### PR TITLE
Add function to print error message and clean up code

### PR DESCRIPTION
**Pull request by KevCui**
In this PR:
1) Add function `print_error` to print error message and exit script after printing
2) Remove duplicated variable declaration of `CUSTOM=${2:-false} `
3) Define global variable `TMPMAIL_API_URL` as API URL
4) Define local variable `regexp` used as regex pattern
5) Make some `if` as one-liner
6) Fix some typos, clean tailing spaces and empty lines

Please review.

### CODE REVIEW

File: tmpmail [hunk 1]
```diff
@@ -103,17 +106,16 @@ generate_email_address() {
...
+        grep -E "$regexp" <<< "$EMAIL_ADDRESS" >/dev/null
+
         # Get the exit status of the command above
         STATUS=$?
 
-        if [ "$STATUS" -ne 0 ]; then
-            echo "Error: Provided email is invalid. Must match [a-z]+@1secmail.(com|net|org)"
-            exit 1
-        fi
+        [[ "$STATUS" -ne 0 ]] \
```
Comment (sdushantha) [inline tmpmail:117]: You can put this into one line, I don't mind if the character length goes over 80 characters. I tend to keep the limit to 100 characters as it seems more reasonable.

File: tmpmail [hunk 2]
```diff
@@ -103,17 +106,16 @@ generate_email_address() {
     if [ "$CUSTOM" != false ]; then
         EMAIL_ADDRESS=$CUSTOM
 
-        # Do a regex check to see if the email address provided by the user is a 
+        # Do a regex check to see if the email address provided by the user is a
         # valid email address
-        echo "$EMAIL_ADDRESS" | grep -E "[a-z]+@1secmail\.(com|net|org)" >/dev/null
-        
+        local regexp="[a-z]+@1secmail\.(com|net|org)"
+        grep -E "$regexp" <<< "$EMAIL_ADDRESS" >/dev/null
```
Comment (sdushantha) [inline tmpmail:112]: Could you make the word. `regexp` uppercase?

I'm not so familiar with the `local` command. Why is it better to define a variable with `local` than without it?

File: tmpmail [hunk 3]
```diff
@@ -103,17 +106,16 @@ generate_email_address() {
...
+
         # Get the exit status of the command above
         STATUS=$?
 
-        if [ "$STATUS" -ne 0 ]; then
-            echo "Error: Provided email is invalid. Must match [a-z]+@1secmail.(com|net|org)"
-            exit 1
-        fi
+        [[ "$STATUS" -ne 0 ]] \
+            && print_error "Provided email is invalid. Must match $regexp"
```
Comment (sdushantha) [inline tmpmail:118]: You can put this into one line, I don't mind if the character length goes over 80 characters. I tend to keep the limit to 100 characters as it seems more reasonable.

File: tmpmail (hunk 2 above)
Comment (sdushantha) [inline tmpmail:112]: Could you make the word. `regexp` uppercase?

I'm not so familiar with the `local` command. Why is it better to define a variable with `local` than without it?

**Review by sdushantha — CHANGES_REQUESTED** [review]

**Comment by sdushantha** [comment]
@KevCui Just so you know, I made`tmpmail` POSIX in the latest version. So some things such as here-strings will not work

File: tmpmail (hunk 3 above)
Comment (KevCui) [reply tmpmail]: Done 4957d3d42313c33c2ee0a32d8ecf83057dc86b4f

File: tmpmail (hunk 2 above)
Comment (KevCui) [reply tmpmail]: Done 4957d3d42313c33c2ee0a32d8ecf83057dc86b4f

**Comment by KevCui** [comment]
Make changes POSIX-compliant, please review :pray:

**Review by sdushantha — APPROVED** [review]

**Comment by sdushantha** [comment]
Thank you for the changes! Everything looks really good :)

**Pull request merged**